File: 167. Two Sum II, Sorted Array/TwoPointerSolution.java
class TwoPointerSolution 
{
    public int[] twoSum(int[] numbers, int target) 
    {
        
        int left = 0;
        int right = numbers.length - 1;

        while (left < right)
        {

            if (numbers[left] + numbers[right] == target)
            {
                return new int[] {left+1, right+1};
            }

            else if (numbers[left] + numbers[right] > target)
            {
                right-- ;
            }

            else
            {
                left++ ;
            }

        }
	
		//This will never get executed but we add it to avoid compilation error as the function has a return type of array.
        return new int[2];
    }
}

File: 167. Two Sum II, Sorted Array/TwoPointerSolution.py
class TwoPointerSolution:

    def twoSum(self, numbers: List[int], target: int) -> List[int]:
       
        length = len(numbers)

        #Initializing left and right pointers
        left = 0
        right = length - 1

        while left < right:

            #Check if the sum is equal to the target
            if numbers[left] + numbers[right] == target:
                return [left+1, right+1]

            #If the sum is greater than the target then shift the right pointer to left
            elif numbers[left] + numbers[right] > target:
                right -= 1

            #If the sum is less than the target then shift the left pointer to right
            else:
                left += 1